### PR TITLE
Map some asynchronous methods with Future

### PR DESCRIPTION
**Pull request by phimage**
using `future`methods discussed in https://github.com/Thomvis/BrightFutures/pull/113
except StoreKit (Using a delegate implementation) and `NSNotificationCenter`(listen block, with automatic observer remove)

Add OSX target
Create a folder for each framework dependency
Add podspec draft (with subspec for each folder)

using head for carthage for the moment

### CODE REVIEW

**Comment by Thomvis** [comment]
Thank you for initiating this! FutureProofing deserves more love.

I'll have a closer look at the PR this weekend and hope to merge it in!

File: FutureProofing/MapKit/MKDirections+Promise.swift [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import MapKit
+#if !COCOAPODS
+import PromiseKit
```
Comment (Thomvis) [inline FutureProofing/MapKit/MKDirections+Promise.swift]: This should be BrightFutures, right?

File: FutureProofing/MapKit/MKMapSnapshotter+Promise.swift [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import MapKit
+#if !COCOAPODS
+import PromiseKit
```
Comment (Thomvis) [inline FutureProofing/MapKit/MKMapSnapshotter+Promise.swift]: This should be BrightFutures, right?

**Comment by Thomvis** [comment]
Is this PR based on https://github.com/mxcl/PromiseKit/tree/master/Categories? I think it is fine, license wise, but I'll add a reference in the README if this is indeed the case.

**Comment by phimage** [comment]
Yes it's inspired from PromiseKit for a part(CloudKit & MapKit, notificationcenter) and forgot to remove this files. And yes you can add a reference
My need has been `NSItemProvider` on OSX, then I have searched on google `completionHandler` to implement other similar methods and see PromiseKit do the same

I commit the files removal

**Comment by toadzky** [comment]
Personally, I think you should avoid depending on `HEAD` if you can avoid it. It allows things to break to easily. Carthage supports semver ranges, so pick a major version and let it resolve the latest compatible one instead of using `HEAD`.

Also, since I'm doing OS X development, I'm wondering how long until this PR is merged so I can use it. Thanks.

**Comment by Thomvis** [comment]
I've merged the PR. Thanks so much for your work!

When pushing the podfile to CocoaPods however I encountered the following error:

```
-> FutureProofing (0.1.0)
    - ERROR | [FutureProofing/CloudKit,FutureProofing/Foundation] xcodebuild: Returned an unsuccessful exit code. You can use `--verbose` for more information.
    - NOTE  | [FutureProofing/CloudKit,FutureProofing/Foundation] xcodebuild:  clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

As such, the release is not yet available via CocoaPods. I don't have time to look into the error any further right now. If you see the problem and have a fix, feel free to do another PR and I'll make sure to merge it in soon.

Thanks again!

**Pull request merged**